pkg/store/consul/consulutil/watch.go

```diff
@@ -56,8 +56,6 @@ func WatchKeys(
 				WaitIndex:  currentIndex,
 				AllowStale: true,
 			})
-			consulLatencyHistogram.Update(int64(queryMeta.RequestTime))
-			listLatencyHistogram.Update(int64(time.Since(listStart) / time.Millisecond))
```

Comment:
this line doesn't depend on `queryMeta`, so it could have stayed, but probably better to do as you have done and keep them together.